### PR TITLE
Add Simulator Control extension

### PR DESCRIPTION
**Pull request by noppefoxwolf**
## Description

`Simulator Control` is `simctl` command wrapper extension for raycast.

## Screencast

CleanShot 2023-01-02 at 10 32 42@2x

CleanShot 2023-01-02 at 10 33 03@2x

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Review by peduarte — COMMENTED** [review]
Thanks for this, will take a look!

**Comment by peduarte** [comment]
When I boot, is it supposed to open the Simulator App?

**Comment by noppefoxwolf** [comment]
@peduarte No, It just launch process. If you want to focus simulator app window, you should to use `Open` action.

File: extensions/simctl/README.md [hunk 1]
```diff
@@ -0,0 +1 @@
+# Simulators
```
Comment (peduarte) [inline extensions/simctl/README.md:1]: Can we add a description here, so it looks nicer on the store? Thanks!

File: extensions/simctl/package.json [hunk 2]
```diff
@@ -0,0 +1,40 @@
+{
+  "$schema": "https://www.raycast.com/schemas/extension.json",
+  "name": "simctl",
+  "title": "Simulator Control",
+  "description": "Control apple simulator devices",
```
Comment (peduarte) [inline extensions/simctl/package.json:5]:
```suggestion
  "description": "Control Apple Simulator devices",
```

File: extensions/simctl/package.json [hunk 3]
```diff
@@ -0,0 +1,40 @@
+{
+  "$schema": "https://www.raycast.com/schemas/extension.json",
+  "name": "simctl",
```
Comment (peduarte) [inline extensions/simctl/package.json:3]: Is the term `simctl` known? Maybe it's just me, but I wonder if having the `name` as `Simulator Control` is clearer.

File: extensions/simctl/package.json [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  "icon": "command-icon.png",
+  "author": "noppefoxwolf",
+  "categories": [
+    "Developer Tools"
+  ],
+  "license": "MIT",
+  "commands": [
+    {
+      "name": "index",
+      "title": "Simctl",
```
Comment (peduarte) [inline extensions/simctl/package.json:15]: Similar comment here, but maybe having `Simulator Control` as the `title` might be clearer. no

File: extensions/simctl/package.json [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+      "description": "Control apple simulator devices",
```
Comment (peduarte) [inline extensions/simctl/package.json:16]:
```suggestion
      "description": "Control Apple Simulator devices",
```

File: extensions/simctl/src/index.tsx [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+    fetchSimulators();
+  }, []);
+
+  const openAction = (device: Device) => {
+    if (device.state !== "Booted") {
+      return null;
+    }
+    return (
+      <Action
+        title="Open"
```
[the remark this answers is not in the document]
Comment (noppefoxwolf) [reply extensions/simctl/src/index.tsx:35]: In fact, I wanted to focus to window of selected device. This action is not as expected now. So I'll fix or omit this feature.

File: extensions/simctl/src/index.tsx [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+          return 0;
+        })
+        .map((device) => {
+          return (
+            <List.Item
+              id={device.udid}
+              title={device.name}
+              key={device.udid}
+              accessories={[
+                { tag: { value: device.state, color: device.state === "Booted" ? Color.Green : Color.Red } },
```
[the remark this answers is not in the document]
Comment (noppefoxwolf) [reply extensions/simctl/src/index.tsx:112]: CleanShot 2023-01-02 at 23 14 21@2x

Looks good!

File: extensions/simctl/package.json (hunk 3 above)
Comment (noppefoxwolf) [reply extensions/simctl/package.json:3]: @peduarte 
Apparently, the package.json's name has recommendation role.
In read this, It's better "simulator-control" than that. right?

https://docs.npmjs.com/cli/v9/configuring-npm/package-json#name

Comment (noppefoxwolf) [inline extensions/simctl/package.json:3]:
```suggestion
  "name": "simulator-control",
```

Comment (peduarte) [reply extensions/simctl/package.json:3]: Yeah `simulator-control` would be good I think!

File: extensions/simctl/src/index.tsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (noppefoxwolf) [reply extensions/simctl/src/index.tsx]: Um, it window managing is difficult using command.
I removed this feature on initial version.

**Comment by noppefoxwolf** [comment]
@peduarte 
Hey, Thank you for your run and writing reviews.
I fixed issues that is feedbacked by you.

**Comment by noppefoxwolf** [comment]
oops! CI is filed. I'll fix it soon.

**Comment by peduarte** [comment]
LGMT! Thanks for your contribution :tad

**Review by peduarte — APPROVED** [review]

**Pull request merged**